src/modules/dmq_usrloc/usrloc_sync.c

```diff
@@ -72,6 +70,7 @@ static int add_contact(str aor, ucontact_info_t* ci)
 		// Search by ruid, if possible
 		res = dmq_ul.get_urecord_by_ruid(_d, dmq_ul.get_aorhash(&aor), &ci->ruid, &r, &c);
```

Comment:
`get_urecord_by_ruid` was also acquiring the domain lock, this is why we must do `lock_udomain` only after it returns